### PR TITLE
Added error when specifying a package icon or license file with the incorrect path casing.

### LINKED ISSUE #9817: License and icon file existence check should always use a case-sensitive comparison
When you attempt to create a NuGet package with a license file that doesn't exist, NuGet will complain with NU5030:

```
The license file 'LICENSE.txt' does not exist in the package.
```

However, [this check uses an OS-dependent string comparison](https://github.com/NuGet/NuGet.Client/blob/1f1213960012d6452b63a267607d1e237318025e/src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs#L588).

This isn't correct as NuGet packages are zip files, which use case-sensitive paths. As a result, using the wrong case for the `license` element results in an invalid package.

--------------

Consider [this modified PackageLicenseFileExample](https://github.com/PathogenPlayground/Samples/blob/master/PackageLicenseFileExample/PackageLicenseFileExample.csproj) from NuGet/Samples. Namely the typo made to `PackageLicenseFile`:

```xml
<PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
```

Running `dotnet pack` results in a "successful" nupkg:

image

Attempting to upload this package to NuGet.org results in an error:

image

If you put the package into a local feed and attempt to view the license from Visual Studio 2019 (16.6.4) you'll get an error:

image

I also tried an old instance of Nexus Package Manager, which did accept the package. (However that may be more due to the fact that it doesn't understand embedded licenses.) I did not check how any other package servers reacted to the broken package.

--------------

## Versions and stuff from the issue template

NuGet product used: `dotnet` and `VS UI`.

NuGet version: Whatever `dotnet` is using under the hood. (The old `nuget.exe` I have kicking around in my environment is 4.8.1.5435)

dotnet.exe --version: 5.0.100-preview.6.20318.15

VS version: 16.6.4

OS version: Windows 10 v1909 (18363.959)

Worked before? Unlikely, Git blame suggests [this check has always worked this way](https://github.com/NuGet/NuGet.Client/pull/2450/files#diff-7d862ed9e52e34d5ee39d6ccbf6f8a7cR494).

--------------

I considered submitting a PR to fix this, but it is unclear whether this change can be made or if it should be a new warning instead to avoid a backwards compatibility issues. (Especially considering the package is still valid to package servers which aren't aware of embedded license files.)

### PR DESCRIPTION
**Pull request by PathogenDavid**
This implements [the design](https://github.com/NuGet/Home/blob/e4141f74ff152b0a28bfdd7a3dca30c6799edf2b/designs/PackageValidation/FileNameCaseSensitivity.md) set forth in https://github.com/NuGet/Home/pull/10049

## Bug

Fixes: https://github.com/NuGet/Home/issues/9817
Regression: No  
* Last working version:   N/A
* How are we preventing it in future:   This PR also [adds a note to `PathUtility.GetStringComparerBasedOnOS` and `PathUtilty.GetStringComparisonBasedOnOS`](https://github.com/PathogenDavid/NuGet.Client/commit/80eaa43dcf2a5fa7106471e1f5dc96c7fb6628b1#diff-a839c2cf45dfebc6afffc9589f537af9bd6d9b12d2bdd0ba1afa600f7aea9313) noting that they should not be used for package paths.

## Fix

Details: The details are in https://github.com/NuGet/Home/blob/e4141f74ff152b0a28bfdd7a3dca30c6799edf2b/designs/PackageValidation/FileNameCaseSensitivity.md

In short: License and package icon validation is now case-sensitive. A case-insensitive check is used to suggest the likely-correct path when the user specifies a path incorrect only by case.

## Testing/Validation

Tests Added: Yes
Reason for not adding tests:  N/A
Validation:  Test suite

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx [hunk 1]
```diff
@@ -212,4 +212,12 @@
     <value>The element 'icon' cannot be empty.</value>
     <comment>Please, don't localize icon</comment>
   </data>
+  <data name="IconNoFileElementWithHint" xml:space="preserve">
+    <value>The icon file '{0}' does not exist in the package. (Did you mean '{1}'?)</value>
+    <comment>0 - the icon file, 1 - hint to guessed icon file</comment>
+  </data>
+  <data name="Manifest_LicenseFileIsNotInNupkgWithHint" xml:space="preserve">
+    <value>The license file '{0}' does not exist in the package. (Did you mean '{1}'?)</value>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx:220]: ❤️

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx [hunk 2]
```diff
@@ -212,4 +212,12 @@
     <value>The element 'icon' cannot be empty.</value>
     <comment>Please, don't localize icon</comment>
   </data>
+  <data name="IconNoFileElementWithHint" xml:space="preserve">
+    <value>The icon file '{0}' does not exist in the package. (Did you mean '{1}'?)</value>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Resources/NuGetResources.resx:216]: ❤️

**Review by nkolev92 — APPROVED** [review]
Looks pretty great to me!

Thanks @PathogenDavid

**Review by erdembayar — APPROVED** [review]

**Comment by PathogenDavid** [comment]
@nkolev92 @erdembayar Thanks for the reviews!

It looks like `End_To_End_Tests_On_Windows Part1` failed, but I can't see the details. Can someone on the NuGet team check them for me?

**Comment by heng-liu** [comment]
Hi @PathogenDavid, the End_To_End_Tests_On_Windows Part1 failed as there is a flaky test. We disabled that flaky test just now. 
If you rebase from dev, all the tests should be good now. Thanks.

**Comment by PathogenDavid** [comment]
@heng-liu Ah, that'd do it, thanks! Just pushed the rebase.

**Review by zivkan — APPROVED** [review]

**Comment by PathogenDavid** [comment]
@zivkan CI seems to have gotten snagged on NuGet.Client-PrivateDev, any insight on why that might be?

**Comment by zivkan** [comment]
transient infrastructure issue 😢    I restarted the build.
